### ISSUE TITLE
[BUG] GH Pages Action broken

### ISSUE THREAD

**Issue by NatalieZelenka** [NatalieZelenka/phenotype_from_genotype#56]
GH Action not updating website :(

Gh-pages peaceiris action is the bit failing:
```
  [INFO] ForceOrphan: false
  /usr/bin/git clone --depth=1 --single-branch --branch gh-pages ***github.com/NatalieThurlby/phenotype_from_genotype.git /home/runner/actions_github_pages_1639098542290
  Cloning into '/home/runner/actions_github_pages_1639098542290'...
  /usr/bin/git rm -r --ignore-unmatch *
  rm '.buildinfo'
  rm '.nojekyll'
  rm '_downloads/0f6968c734eb0e3592fbb93afaaafdcf/make_dcgo_input.py'
  rm '_downloads/1d5aed2e2ac99c61648969ced4c4bf8c/create-base-simulated-counts.R'
  rm '_downloads/21b06bbc87849cd431043497ef629dae/fasta_to_uniprot.py'
  rm '_downloads/413ee71f781407eaac54a462330a3b01/download-combined.R'
  rm '_images/alignment.png'
  rm '_images/amino_acid_ribosome.png'
  rm '_images/blank.png'
  rm '_images/bristol-alt-crest-red.png'
  rm '_images/bristol-crest.png'
  rm '_images/classifies-people.png'
  rm '_images/clustering_comparison.png'
  rm '_images/clustering_snps.png'
  rm '_images/curse-of-dimensionality.png'
  rm '_images/de_novo_assembly.png'
  rm '_images/difficult-to-understand.png'
  rm '_images/dna-both.png'
  rm '_images/filip.png'
  rm '_images/filip_bootstrap.png'
  rm '_images/general-hazard.png'
  rm '_images/go_rilla.png'
  rm '_images/interesting-scores.png'
  rm '_images/lacks-community.png'
  rm '_images/lego.png'
  rm '_images/linear_metric.png'
  rm '_images/linneaus_ehret.png'
  rm '_images/mendel.png'
  rm '_images/misuse.png'
  rm '_images/mycobacterium_tuberculosis_Yyz.png'
  rm '_images/myobacterium_tuberculosis_rKb.png'
  rm '_images/nonlinear_metric.png'
  rm '_images/ontolopy_logo.png'
  rm '_images/p_hacking.png'
  rm '_images/pca_real.png'
  rm '_images/privacy.png'
  rm '_images/reinforce-bias.png'
  rm '_images/revigo_filip_wrong_cafa2.png'
  rm '_images/shaded_score.png'
  rm '_images/snowflake-overview-new.png'
  rm '_images/tissues_HPA.png'
  rm '_images/transcription.png'
  rm '_images/venn_brain.png'
  rm '_panels_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css'
  rm '_panels_static/panels-variables.06eb56fa6e07937060861dad626602ad.css'
  rm '_sources/c0-front-matter/01-title-page.md'
  rm '_sources/c0-front-matter/02-declaration.md'
  rm '_sources/c0-front-matter/03-abstract.md'
  rm '_sources/c0-front-matter/04-acknowledgements.md'
  rm '_sources/c0-front-matter/05-full-table-of-contents.md'
  rm '_sources/c01-introduction/intro.md'
  rm '_sources/c02-biology-background/0-index.md'
  rm '_sources/c02-biology-background/1-big-questions.md'
  rm '_sources/c02-biology-background/2-biological-molecules.md'
  rm '_sources/c02-biology-background/3-more-dna.md'
  rm '_sources/c02-biology-background/4-more-proteins.md'
  rm '_sources/c02-biology-background/5-phenotype.md'
  rm '_sources/c02-biology-background/6-summary.md'
  rm '_sources/c03-compbio-background/0-index.md'
  rm '_sources/c03-compbio-background/1-sequencing-technology.md'
  rm '_sources/c03-compbio-background/2-measuring-genotype-phenotype.md'
  rm '_sources/c03-compbio-background/3-ontologies.md'
  rm '_sources/c03-compbio-background/4-comp-bio-methods.md'
  rm '_sources/c03-compbio-background/5-bias.md'
  rm '_sources/c03-compbio-background/6-pqi.md'
  rm '_sources/c03-compbio-background/7-summary.md'
  rm '_sources/c04-snowflake/0-index.md'
  rm '_sources/c04-snowflake/1-introduction.md'
  rm '_sources/c04-snowflake/2-snowflake-algorithm.md'
  rm '_sources/c04-snowflake/3-creating-inputs.md'
  rm '_sources/c04-snowflake/4-preprocessing.md'
  rm '_sources/c04-snowflake/5-clustering-snps.md'
  rm '_sources/c04-snowflake/7-discussion.md'
  rm '_sources/c05-alspac/0-index.md'
  rm '_sources/c05-alspac/1-introduction.md'
  rm '_sources/c05-alspac/5-discussion.md'
  rm '_sources/c06-filter/0-index.md'
  rm '_sources/c06-filter/1-introduction.md'
  rm '_sources/c06-filter/2-algorithm.md'
  rm '_sources/c06-filter/3-data.md'
  rm '_sources/c06-filter/5-methods.md'
  rm '_sources/c06-filter/6-results.md'
  rm '_sources/c06-filter/7-discussion.md'
  rm '_sources/c07-ontolopy/0-index.md'
  rm '_sources/c07-ontolopy/1-introduction.md'
  rm '_sources/c07-ontolopy/2-functionality.md'
  rm '_sources/c07-ontolopy/3-how-it-works.md'
  rm '_sources/c07-ontolopy/4-misc-examples.md'
  rm '_sources/c07-ontolopy/5-mapping-example.md'
  rm '_sources/c07-ontolopy/6-discussion.md'
  rm '_sources/c07-ontolopy/7-future-work.md'
  rm '_sources/c08-combining/0-index.md'
  rm '_sources/c08-combining/1-background.md'
  rm '_sources/c08-combining/2-data.md'
  rm '_sources/c08-combining/3-data-wrangling.md'
  rm '_sources/c08-combining/7-discussion.md'
  rm '_sources/c09-conclusion/0-conclusion.md'
  rm '_sources/cz-end-matter/0-appendix.md'
  rm '_sources/cz-end-matter/reference.md'
  rm '_sources/jupyter_book_intro.md'
  rm '_static/__init__.py'
  rm '_static/__pycache__/__init__.cpython-37.pyc'
  rm '_static/basic.css'
  rm '_static/check-solid.svg'
  rm '_static/clipboard.min.js'
  rm '_static/combining_funnel_interactive.html'
  rm '_static/copy-button.svg'
  rm '_static/copybutton.css'
  rm '_static/copybutton.js'
  rm '_static/copybutton_funcs.js'
  rm '_static/css/index.c5995385ac14fb8791e8eb36b4908be2.css'
  rm '_static/css/theme.css'
  rm '_static/custom.css'
  rm '_static/doctools.js'
  rm '_static/documentation_options.js'
  rm '_static/favicon.png'
  rm '_static/file.png'
  rm '_static/images/logo_binder.svg'
  rm '_static/images/logo_colab.png'
  rm '_static/images/logo_jupyterhub.svg'
  rm '_static/jquery-3.5.1.js'
  rm '_static/jquery.js'
  rm '_static/js/index.1c5a1a01449ed65a7b51.js'
  rm '_static/language_data.js'
  rm '_static/minus.png'
  rm '_static/mystnb.css'
  rm '_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css'
  rm '_static/panels-variables.06eb56fa6e07937060861dad626602ad.css'
  rm '_static/plus.png'
  rm '_static/pygments.css'
  rm '_static/searchtools.js'
  rm '_static/sphinx-book-theme.12a9622fbb08dcb3a2a40b2c02b83a57.js'
  rm '_static/sphinx-book-theme.css'
  rm '_static/sphinx-book-theme.e8f53015daec13862f6db5e763c41738.css'
  rm '_static/sphinx-thebe.css'
  rm '_static/sphinx-thebe.js'
  rm '_static/thesis.pdf'
  rm '_static/thesis_logo_with_text.png'
  rm '_static/togglebutton.css'
  rm '_static/togglebutton.js'
  rm '_static/underscore-1.12.0.js'
  rm '_static/underscore.js'
  rm '_static/vendor/fontawesome/5.13.0/LICENSE.txt'
  rm '_static/vendor/fontawesome/5.13.0/css/all.min.css'
  rm '_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.eot'
  rm '_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.svg'
  rm '_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.ttf'
  rm '_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff'
  rm '_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2'
  rm '_static/vendor/fontawesome/5.13.0/webfonts/fa-regular-400.eot'
  rm '_static/vendor/fontawesome/5.13.0/webfonts/fa-regular-400.svg'
  rm '_static/vendor/fontawesome/5.13.0/webfonts/fa-regular-400.ttf'
  rm '_static/vendor/fontawesome/5.13.0/webfonts/fa-regular-400.woff'
  rm '_static/vendor/fontawesome/5.13.0/webfonts/fa-regular-400.woff2'
  rm '_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.eot'
  rm '_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.svg'
  rm '_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.ttf'
  rm '_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff'
  rm '_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2'
  rm '_static/webpack-macros.html'
  rm 'c0-front-matter/01-title-page.html'
  rm 'c0-front-matter/02-declaration.html'
  rm 'c0-front-matter/03-abstract.html'
  rm 'c0-front-matter/04-acknowledgements.html'
  rm 'c0-front-matter/05-full-table-of-contents.html'
  rm 'c01-introduction/intro.html'
  rm 'c02-biology-background/0-index.html'
  rm 'c02-biology-background/1-big-questions.html'
  rm 'c02-biology-background/2-biological-molecules.html'
  rm 'c02-biology-background/3-more-dna.html'
  rm 'c02-biology-background/4-more-proteins.html'
  rm 'c02-biology-background/5-phenotype.html'
  rm 'c02-biology-background/6-summary.html'
  rm 'c03-compbio-background/0-index.html'
  rm 'c03-compbio-background/1-sequencing-technology.html'
  rm 'c03-compbio-background/2-measuring-genotype-phenotype.html'
  rm 'c03-compbio-background/3-ontologies.html'
  rm 'c03-compbio-background/4-comp-bio-methods.html'
  rm 'c03-compbio-background/5-bias.html'
  rm 'c03-compbio-background/6-pqi.html'
  rm 'c03-compbio-background/7-summary.html'
  rm 'c04-snowflake/0-index.html'
  rm 'c04-snowflake/1-introduction.html'
  rm 'c04-snowflake/2-snowflake-algorithm.html'
  rm 'c04-snowflake/3-creating-inputs.html'
  rm 'c04-snowflake/4-preprocessing.html'
  rm 'c04-snowflake/5-clustering-snps.html'
  rm 'c04-snowflake/7-discussion.html'
  rm 'c05-alspac/0-index.html'
  rm 'c05-alspac/1-introduction.html'
  rm 'c05-alspac/5-discussion.html'
  rm 'c06-filter/0-index.html'
  rm 'c06-filter/1-introduction.html'
  rm 'c06-filter/2-algorithm.html'
  rm 'c06-filter/3-data.html'
  rm 'c06-filter/5-methods.html'
  rm 'c06-filter/6-results.html'
  rm 'c06-filter/7-discussion.html'
  rm 'c07-ontolopy/0-index.html'
  rm 'c07-ontolopy/1-introduction.html'
  rm 'c07-ontolopy/2-functionality.html'
  rm 'c07-ontolopy/3-how-it-works.html'
  rm 'c07-ontolopy/4-misc-examples.html'
  rm 'c07-ontolopy/5-mapping-example.html'
  rm 'c07-ontolopy/6-discussion.html'
  rm 'c07-ontolopy/7-future-work.html'
  rm 'c08-combining/0-index.html'
  rm 'c08-combining/1-background.html'
  rm 'c08-combining/2-data.html'
  rm 'c08-combining/3-data-wrangling.html'
  rm 'c08-combining/7-discussion.html'
  rm 'c09-conclusion/0-conclusion.html'
  rm 'cz-end-matter/0-appendix.html'
  rm 'cz-end-matter/reference.html'
  rm 'genindex.html'
  rm 'index.html'
  rm 'jupyter_book_intro.html'
  rm 'objects.inv'
  rm 'search.html'
  rm 'searchindex.js'
  [INFO] first deployment, create new branch gh-pages
  ENOENT: no such file or directory, scandir '/home/runner/work/phenotype_from_genotype/phenotype_from_genotype/_build/html'
  /usr/bin/git init
  Reinitialized existing Git repository in /home/runner/actions_github_pages_1639098542290/.git/
  /usr/bin/git checkout --orphan gh-pages
  fatal: A branch named 'gh-pages' already exists.
  Error: Action failed with "The process '/usr/bin/git' failed with exit code 128"
```

**Comment by NatalieThurlby** [comment]
- Updated version of peaceiris action: now works but deleted website :scream:

**Comment by NatalieZelenka** [comment]
New similar issue. V1 of the R GitHub action broke. Changed to V2.

**Comment by NatalieZelenka** [comment]
New error: jinja2 4.x incompatible with sphinx 3.x
Fixed using suggestion (1) from here: https://github.com/executablebooks/jupyter-book/issues/1676 
Suggestion (2) doesn't work with this version of jupyter-book.

**Comment by NatalieZelenka** [comment]
FINALLY fixed!! (having gone through quite a lot of dependency problems) - latest fix was updating to R 4.0.

**Comment by NatalieZelenka** [comment]
Note - the dependency problems I originally attempted to fix by installing specific versions of R packages, but this turned out to be unnecessary.

Leaving here just in case:
```R

IRkernel = "https://cran.r-project.org/package=IRkernel&version=1.1.1"
glue = "https://cran.r-project.org/package=glue&version=1.2.0"

utils::install.packages(pkgs = glue, repos = NULL)

install.packages('pillar')
install.packages('repr')
install.packages('evaluate')
install.packages('IRdisplay')
install.packages('pbdZMQ')
install.packages('crayon')
install.packages('jsonlite')
install.packages('uuid')
install.packages('digest')

utils::install.packages(pkgs = IRkernel, repos = NULL)

```
